src/charge_controller/dev/ControllerUI.cpp

```diff
@@ -0,0 +1,106 @@
+#include <charge_controller/dev/ControllerUI.hpp>
+
+ControllerUI::ControllerUI(DEV::Encoder& encoder, DEV::Button& encoderButton, LCDDisplay& display, ControllerModel& model) : model(model), encoder(encoder), encoderButton(encoderButton), display(display) {
+    encoder.setRangeAndPosition(model.getEncoderRange(ControllerModel::State::PAGE_SELECT), 0);
+}
+
+void ControllerUI::process() {
+    switch (model.getState()) {
+    case ControllerModel::State::PAGE_SELECT:
+        switch (encoder.getPosition()) {
+        case ControllerModel::Page::MAIN:
+            model.setPage(ControllerModel::Page::MAIN);
+            break;
+        case ControllerModel::Page::SETTINGS:
+            model.setPage(ControllerModel::Page::SETTINGS);
+            break;
+        default:
+            //THIS SHOULD NEVER HAPPEN
```

Comment:
This can just be a log statement for now. If this were a mission-critical system, I'd like some more advanced error handling, but this isn't a huge deal

Reply:
May be worth adding a Trello card for advanced error handling